app/models/concerns/foreman_ansible/host_managed_extensions.rb

```diff
@@ -22,14 +23,13 @@ def inherited_ansible_roles
 
       def play_ansible_roles
         return unless ansible_roles.present? || inherited_ansible_roles.present?
-        task = ::ForemanTasks.async_task(
-          ::Actions::ForemanAnsible::PlayHostRoles,
-          self,
-          ::ForemanAnsible::ProxySelector.new,
-          :timeout => Setting['ansible_post_provision_timeout']
-        )
+        composer = job_composer(:ansible_run_host, self)
+        composer.trigger!
         logger.info("Task for Ansible roles on #{self} before_provision: "\
-                    "#{Rails.application.routes.url_helpers.task_path(task)}.")
+                    "#{job_invocation_path(composer.job_invocation)}")
+      rescue Foreman::Exception => e
+        logger.info("Error running Ansible roles on #{self} before_provision: "\
```

Comment:
not in this commit, but I'm thinking we should have "error" notification blueprint in core and this should render such notification, another example would be, facts parsing failed...

Reply:
Just as a reminder to myself, audience for that kind of notification should be `host.owner`